### PR TITLE
PDCL-12483 Add conflict resolution support for In-App messages.

### PR DESCRIPTION
**Pull request by dompuiu**
## Description

This PR is rendering on the page the In-App message with the highest rank. For the others that are not shown, a suppressDisplay event is sent back.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [x] I have run the Sandbox successfully.

### CODE REVIEW

File: src/components/Personalization/createOnDecisionHandler.js [hunk 1]
```diff
@@ -10,6 +10,22 @@ OF ANY KIND, either express or implied. See the License for the specific languag
...
+  let count = 0;
+  return (proposition) => {
+    const { items = [] } = proposition;
+
+    if (!items.some((item) => item.schema === MESSAGE_IN_APP)) {
+      return false;
+    }
+    count += 1;
+
+    return count > 1;
```
Comment (carterworks) [inline src/components/Personalization/createOnDecisionHandler.js:25]: nitpick: a comment here on why we only suppress display after there has been at least 2 `items` with no in-app messages would be extremely helpful.

File: src/components/Personalization/createNotificationHandler.js [hunk 2]
```diff
@@ -24,10 +25,25 @@ export default (collect, renderedPropositions) => {
       return renderedPropositionsDeferred.resolve;
     }
 
-    return (decisionsMeta) => {
-      if (decisionsMeta.length > 0) {
+    return (decisionsMetaDisplay = [], decisionsMetaSuppressed = []) => {
+      if (
+        Array.isArray(decisionsMetaDisplay) &&
```
Comment (ninaceban) [inline src/components/Personalization/createNotificationHandler.js:30]: we have an util that does both checks - `isNonEmptyArray`

File: src/components/Personalization/createNotificationHandler.js [hunk 3]
```diff
@@ -24,10 +25,25 @@ export default (collect, renderedPropositions) => {
...
         collect({
-          decisionsMeta,
+          decisionsMeta: decisionsMetaDisplay,
+          viewName,
+        });
+      }
+
+      if (
+        Array.isArray(decisionsMetaSuppressed) &&
+        decisionsMetaSuppressed.length > 0
```
Comment (ninaceban) [inline src/components/Personalization/createNotificationHandler.js:41]: here as well

File: src/components/Personalization/createNotificationHandler.js (hunk 2 above)
Comment (dompuiu) [reply src/components/Personalization/createNotificationHandler.js]: Changed.

File: src/components/Personalization/createNotificationHandler.js (hunk 3 above)
Comment (dompuiu) [reply src/components/Personalization/createNotificationHandler.js]: Changed.

**Pull request merged**